daemon/logger/syslog/syslog.go

```diff
@@ -0,0 +1,61 @@
+package syslog
+
+import (
+	"log/syslog"
+	"os"
+	"path"
+	"sync"
+
+	"github.com/docker/docker/daemon/logger"
+)
+
+type Syslog struct {
+	writer *syslog.Writer
+	tag    string
+	mu     sync.Mutex
+}
+
+func New(tag string) (logger.Logger, error) {
+	return &Syslog{
+		writer: nil,
+		tag:    tag,
+	}, nil
+}
+
+func (s *Syslog) Log(msg *logger.Message) error {
+	s.mu.Lock()
```

Comment:
The lazy initialization was because if you failed on syslog.New you couldn't create a container.  Seemed a bit much to fail starting containers because syslog wasn't up or having issues.

@crosbymichael Following your logic though, can't we just lock if you need to do syslog.New only?